ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -13972,6 +13977,32 @@ class SemanticErrorChecker : public QuickHqlTransformer
     HqlExprCopyArray locations;
 };
 
+void SemanticErrorChecker::checkBloom(IHqlExpression * bloom)
+{
+    // Possible semantic errors caught here:
+    // 1. Args to active, limit, or probability sub-attrs not constant
+    // 2. Arg to probability out of range
+    ForEachChildFrom(i, bloom, 1)
+    {
+        IHqlExpression * cur = bloom->queryChild(i);
+        const IAtom *name = cur->queryName();
+        if (name==activeAtom)
+            name = bloomAtom;
+        else if (name==probabilityAtom)
+        {
+            IValue *cval = cur->queryChild(0)->queryValue();
+            if (cval)
+            {
+                double rval = cval->getRealValue();
+                if (rval <= 0.01 || rval > 0.3)
+                    reportError(ERR_INVALID_PROBABILITY,"Probability argument is not in permitted range 0.01 to 0.3");
+            }
+        }
+        if (!cur->isConstant())
```

Comment:
Is this desirable/required?

Reply:
Yes. Otherwise the bloomInfo objects have to be able to support onCreate, engines need to call it, info necessary needs to be serialized to slaves, etc etc. I don't want to implement any of that.

Reply:
ok.  It could be relaxed later if it became a requirement.